src/app.js

```diff
@@ -0,0 +1,247 @@
+const BASE_URL = "https://newsapi.org/v1/articles?source=";
+const channelUILists = document.querySelectorAll('.thumbnail');
+const newsSection = document.querySelector('.newsSection');
+/* import { API_KEY as key } from './variables.js'; //Import and Rename
+console.log(key); */
+
+//Proxy usage to get all the Author names
+const authorNames = () => {
+    let validator = {
+        get(target, key) {
+            return key in target ? taget[key] : "No Author";
+        }
+    }
+    let authors = new Proxy({}, validator);
+}
+
+const settings = {
+  spinnerSelector: '.spinner',
+  ulSelector: '#newsArticlesList',
+  gridSection: '.newsSection',
+  channelUIListsSelector: '.thumbnail',
+  BASE_URL,
+  API_KEY
+}
+
+const sources = [
+  {
+    source: 'CNBC',
+    label: 'CNBC News',
+    inst: 'cnbc'
+  },
+  {
+    source: 'Time',
+    label: 'Time News',
+    inst: 'time'
+  },
+  {
+    source: 'usa-today',
+    label: 'Usa Today News',
+    inst: 'usaToday'
+  },
+  {
+    source: 'the-new-york-times',
+    label: 'The New York Times News',
+    inst: 'newYourkTimes'
+  },
+  {
+    source: 'cnn',
+    label: 'CNN News',
+    inst: 'cnn'
+  },
+
+  {
+    source: 'associated-press',
+    label: 'Associated Press News',
+    inst: 'associatedPress'
+  }
+]
+
+class Spinner {
+  constructor(selector) {
+    this.spinner = document.querySelector(selector)
+    this.spinner.style.display = 'none';
+  }
+
+  enableSpinner() {
+    this.spinner.style.display = 'block';
+  }
+
+  disableSpinner() {
+    this.spinner.style.display = 'none';
+  }
+
+}
+
+class App {
+  constructor (sources, settings) {
+    this.sources = sources
+    this.settings = settings
+    this.spinner = new Spinner(settings.spinnerSelector)
+    this.main = document.querySelector(settings.gridSection)
+    this.ul = document.querySelector(settings.ulSelector)
+    this.channelUILists = document.querySelector(settings.channelUIListsSelector)
+    //this.anchor = document.querySelector(settings.buttonSelector)
+    this.channels = {}
+  }
+
+  append(parent, el) {
+    return parent.appendChild(el)
+  }
+
+  createNode(element) {
+    return document.createElement(element)
+  }
+
+  get(source) {
+    this.spinner.enableSpinner()
+  
+    return fetch(`${BASE_URL}${source}&apiKey=${API_KEY}`)
+        .then(response => response.json())
+    }   
+    
+  handleMainClick({ target }) {
+      if(target.tagName.toLowerCase() === 'a' && target.dataset.loadChannel) {
+        const channelName = target.dataset.loadChannel;
+        console.log(channelName);
+        this.channels[channelName].fetchChannel()
+      }
+      this.main.style.display = 'none';
+    }
+
+  init() {  
+    this.channels = this.sources.reduce((acc, { source, inst }) => {
+      acc[inst] = new NewsChannel(source, this);
+      return acc
+    }, {})
+    
+    this.main.addEventListener('click', this.handleMainClick.bind(this))    
+    this.sources.forEach((item)=>{
+      this.main.innerHTML +=
```

Comment:
That could be moved to the separate method

Reply:
Created separate initialContent method